src/Containers/Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -46,25 +46,66 @@ private static bool TryParseAuthenticationInfo(HttpResponseMessage msg, [NotNull
         if (header is { Scheme: "Bearer" or "Basic", Parameter: string bearerArgs })
         {
             scheme = header.Scheme;
-            Dictionary<string, string> keyValues = new();
-            foreach (Match match in BearerParameterSplitter().Matches(bearerArgs))
+            var keyValues = ParseBearerArgs(bearerArgs);
+
+            var result = scheme switch
             {
-                keyValues.Add(match.Groups["key"].Value, match.Groups["value"].Value);
-            }
+                "Bearer" => TryParseBearerAuthInfo(keyValues, out authInfo),
+                "Basic" => TryParseBasicAuthInfo(keyValues, msg.RequestMessage!.RequestUri!, registry, out authInfo),
+                _ => false
+            };
+            return result;
+        }
+        return false;
 
-            if (keyValues.TryGetValue("realm", out string? realm) && keyValues.TryGetValue("service", out string? service))
+        static bool TryParseBearerAuthInfo(Dictionary<string, string> authValues, [NotNullWhen(true)] out AuthInfo? authInfo) {
+            if (authValues.TryGetValue("realm", out string? realm) && authValues.TryGetValue("service", out string? service))
             {
```

Comment:
```suggestion
            if (authValues.TryGetValue("realm", out string? realm))
            {
                string? service = null;
                keyValues.TryGetValue("service", out service);
```

Reply:
This fixes an issue with the OpenShift internal registry which doesn't include a `service`.